framework/src/org/checkerframework/framework/util/AtmLubVisitor.java

```diff
@@ -115,14 +112,8 @@ private AnnotatedTypeMirror lubWithNull(
     private void lubPrimaryAnnotations(
             AnnotatedTypeMirror type1, AnnotatedTypeMirror type2, AnnotatedTypeMirror lub) {
         Set<? extends AnnotationMirror> lubSet;
-        if (visitingUninferedWildcard
-                        && type1.getAnnotations().size() != type1.getAnnotations().size()
```

Comment:
This condition uses type1 twice, making it always false, making the whole first clause false, and effectively only leaving the `type1.getAnnotations().isEmpty()`.
Then, because type1 has no annotations, the nested condition can also be simplified.
As this was the only place that used visitingUninferedWildcard, I also removed that variable and the helper method.
@smillst Does this change look ok to you?